### DISCUSSION TITLE
Visualization of model evaluation (image, true mask, predicted mask)

### DISCUSSION THREAD

**Discussion by HripsimeS** [Project-MONAI/MONAI#3911] [Q&A] [answered]
Hello. I am trying to do a visualization after the evaluation on original images/masks. For example I have 10 images/masks that I would like the test my model and then would like to plot results of all of them with images, true masks and predicted masks. The solution below does not work efficiently as it is plotting only one case (image, true mask and predicted mask) multiple times. Let me know if you have more efficient solution for this visualization. Look forward to hearing from you.

loader = LoadImage()
original_image = loader(test_data[0]["image_meta_dict"]["filename_or_obj"])[0]

with torch.no_grad():
for i, test_data in enumerate(test_loader):

    plt.figure("check", (18, 6))
    
    plt.subplot(1, 3, 1)
    plt.title(f"image {i}")
    plt.imshow(original_image[:, :, 250], cmap="gray")
    
    plt.subplot(1, 3, 2)
    plt.title(f"mask {i}")
    plt.imshow(test_labels[0].detach().cpu()[1, :, :, 250])
    
    plt.subplot(1, 3, 3)
    plt.title(f"prediction {i}")
    plt.imshow(test_outputs[0].detach().cpu()[1, :, :, 250])
    
    plt.show()
    if i == 2:
        break

P.S. If for evaluation of model I am using 10 images/masks, the metric_org is returning the average of all 10 scores? If yes, then how I can check dice score of each case individually? This is to help the results becomes more explainable. This is the target part:

dice_metric(y_pred=test_outputs, y=test_labels)
metric_org = dice_metric.aggregate().item()
dice_metric.reset()

**Comment by Nic-Ma** [comment, marked as answer]
Hi @yiheng-wang-nv ,

Could you please help with this ticket? I think you did lots of visualization tests in our tutorials.

Thanks in advance.

**Reply by HripsimeS** [reply]
@Nic-Ma thanks for your reply! @yiheng-wang-nv already kindly answered to my ticket on tutorial discussions and I could succeed to have nice plots eventually :) Thanks to the awesome MONAI  team 🥇 👍 

https://github.com/Project-MONAI/tutorials/discussions/607
